scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Connection.java

```diff
@@ -282,14 +282,17 @@ public void updatePeerAddress(InetSocketAddress peerAddress) {
 			this.peerAddress = peerAddress;
 			if (establishedSession != null) {
 				establishedSession.setPeer(peerAddress);
-			} else if (peerAddress == null) {
+			} else if (peerAddress != null) {
```

Comment:
I don't get it. If a none-null address is applied without established session (e.g. accidentality, if the address is changed during handshake), this still throws an exception.
A ongoing handshake is failed, when null gets applied.
The differences is, that before no listener is called, if the connection has a established session, now the listeners are even called, if the connection has a established session, but that's different from the one of the handshake.  The javadoc is not that precise here, so it is not changed.

Reply:
Maybe I missed something but :

```If the new address is {@code null}, an ongoing handshake is failed```

sounds not true anymore as for a close notify there is not an ongoing handshake failure involved.

Reply:
I still not clear, what is wrong.

```
error = new HandshakeException("Received 'close notify'", alert);
if (handshaker != null) {
	handshaker.setFailureCause(error);
}
```
So, for close, we fail the handshake, if there is one. if the address is reset and there is a ongoing handshake we fail that handshake.
 
> as for a close notify there is not an ongoing handshake failure involved.

Maybe, this could be implemented in an other way, but as far as I know, we fail handshakes, if close is received.

Reply:
If you still feel, that the javadoc should be corrected, we can do this after the merge.